### DISCUSSION TITLE
How to use CodeBlock component in custom MDX loader?

### DISCUSSION THREAD

**Discussion by phuctm97** [facebook/docusaurus#4438] [Q&A] [answered]
Hi everyone, I'm writing a plugin that allows importing MDX documents as JSX components, everything works except that code block seems not being used, here is my webpack config:

```js
 configureWebpack(config, isServer, utils) {
    const { getCacheLoader, getBabelLoader } = utils;
    return {
      module: {
        rules: [
          {
            test: /\.mdx$/,
            include: [path.resolve(context.siteDir, options.rootDir)],
            use: [
              getCacheLoader(isServer),
              getBabelLoader(isServer),
              require.resolve("@docusaurus/mdx-loader"),
            ],
          },
        ],
      },
    };
  },
```

I tried to look at docusaurus code but didn't find anywhere that handles the logic to use CodeBlock and prism.

Thank you 🙏🏻

**Comment by slorber** [comment, marked as answer]
The `@theme/CodeBlock` component is provided by the classic theme and we have the `MDXProvider` that tells MDX to use it for code  blocks instead of the regular HTML `<code>` element.

You probably don't apply the MDX provider in your code

```jsx
<MDXProvider components={MDXComponents}>{children}</MDXProvider>
```

**Reply by phuctm97** [reply]
Thank you @slorber! Works perfectly
